interface/patient_file/encounter/find_code_dynamic_ajax.php

```diff
@@ -0,0 +1,333 @@
+       ),
+  array('field_id'     => 'reason',
+        'title'        => xl('Reason for Visit'),
+        'uor'          => '1',
+        'data_type'    => '2',               // Text
+        'list_id'      => '',
+        'edit_options' => '',
+       ),
+  array('field_id'     => 'onset_date',
+        'title'        => xl('Date of Onset'),
+        'uor'          => '1',
+        'data_type'    => '4',               // Text-date
+        'list_id'      => '',
+        'edit_options' => '',
+       ),
+  array('field_id'     => 'referral_source',
+        'title'        => xl('Referral Source'),
+        'uor'          => '1',
+        'data_type'    => '1',               // List
+        'list_id'      => 'refsource',
+        'edit_options' => '',
+       ),
+  array('field_id'     => 'shift',
+        'title'        => xl('Shift'),
+        'uor'          => '1',
+        'data_type'    => '1',               // List
+        'list_id'      => 'shift',
+        'edit_options' => '',
+       ),
+  array('field_id'     => 'billing_facility',
+        'title'        => xl('Billing Facility'),
+        'uor'          => '1',
+        'data_type'    => '35',              // Facilities
+        'list_id'      => '',
+        'edit_options' => '',
+       ),
+  array('field_id'     => 'voucher_number',
+        'title'        => xl('Voucher Number'),
+        'uor'          => '1',
+        'data_type'    => '2',               // Text
+        'list_id'      => '',
+        'edit_options' => '',
+       ),
+);
+
+function feSearchSort($search='', $column=0, $reverse=false) {
+  global $form_encounter_layout;
+  $arr = array();
+  foreach ($form_encounter_layout as $feitem) {
+    if ($search && stripos($feitem['field_id'], $search) === false &&
+      stripos($feitem['title'], $search) === false ) {
+      continue;
+    }
+    $feitem['fld_length' ] = 20;
+    $feitem['max_length' ] = 0;
+    $feitem['titlecols'  ] = 1;
+    $feitem['datacols'   ] = 3;
+    $feitem['description'] = '';
+    $feitem['fld_rows'   ] = 0;
+    $key = $column ? 'title' : 'field_id';
+    $arr[$feitem[$key]] = $feitem;
+  }
+  ksort($arr);
+  if ($reverse) $arr = array_reverse($arr);
+  return $arr;
+}
+
+function genFieldIdString($row) {
+  return 'CID|' . json_encode($row);
+}
+
+// Column sorting parameters.
+//
+$orderby = '';
+$fe_column = 0;
+$fe_reverse = false;
+if (isset($_GET['iSortCol_0'])) {
+	for ($i = 0; $i < intval($_GET['iSortingCols']); ++$i) {
+    $iSortCol = intval($_GET["iSortCol_$i"]);
+		if ($_GET["bSortable_$iSortCol"] == "true" ) {
+      $sSortDir = escape_sort_order($_GET["sSortDir_$i"]); // ASC or DESC
+      // We are to sort on column # $iSortCol in direction $sSortDir.
+      $orderby .= $orderby ? ', ' : 'ORDER BY ';
+
+      // Note the primary sort column and direction for later logic.
+      if ($i == 0) {
+        $fe_column = $iSortCol;
+        $fe_reverse = $sSortDir == 'DESC';
+      }
+
+      if ($what == 'codes') {
```

Comment:
Are these for codes that aren't external type (suchs as ICD9/ICD10/SNOMED etc.)

Reply:
Yeah that can be a problem. I'll fix it.